gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CredentialServiceImpl.java

```diff
@@ -140,7 +148,27 @@ public Completable delete(String id) {
         LOGGER.debug("Delete credential {}", id);
         return credentialRepository.findById(id)
                 .switchIfEmpty(Maybe.error(new CredentialNotFoundException(id)))
-                .flatMapCompletable(email -> credentialRepository.delete(id))
+                .flatMapCompletable(credential -> Completable.create(emitter -> {
+                    userService.findById(credential.getUserId()).subscribe(
```

Comment:
We can probably simplify this `flatMapCompletable` by using something like `FlatMap` on the `userService.findById` to avoid multiple call to `subscribe` mehod.

Something like this (did test it)

```
.flatMapCompletable(credential -> 
                    userService.findById(credential.getUserId()).flatMapCompletable(user -> {
            Optional<EnrolledFactor> fido2Factor = user.getFactors()
                    .stream()
                    .filter(enrolledFactor ->
                            FactorSecurityType.WEBAUTHN_CREDENTIAL.equals(enrolledFactor.getSecurity().getType()) &&
                                    enrolledFactor.getSecurity().getValue().equals(credential.getCredentialId())
                    )
                    .findFirst();
            if (fido2Factor.isPresent()) {
                return Completable.error()
            } else {
                return credentialRepository.delete(id);
            }
        } )
```